sigstore/sign.py

```diff
@@ -89,86 +89,105 @@ class Signer:
     The primary API for signing operations.
     """
 
-    def __init__(self, identity_token: str, cache: bool = True) -> None:
+    def __init__(
+        self, identity_token: str, fulcio: FulcioClient, cache: bool = True
+    ) -> None:
         """
         Create a new `Signer`.
 
         `identity_token` is the identity token used to request a signing certificate
         from Fulcio.
 
+        `fulcio` is a `FulcioClient` capable of connecting to a Fulcio instance
+        and returning signing certificates.
+
         `cache` determines whether the signing certificate and ephemeral private key
         should be reused (until the certificate expires) to sign different artifacts.
         Default is `True`.
         """
         self._identity_token: str = identity_token
-        self.cache: bool = cache
-        self._signing_certificate: Optional[FulcioCertificateSigningResponse] = None
-        self._private_key: Optional[ec.EllipticCurvePrivateKey] = None
+        self.__cached_private_key: Optional[ec.EllipticCurvePrivateKey] = None
+        self.__private_key: Optional[ec.EllipticCurvePrivateKey] = None
+        self.__cached_signing_certificate: Optional[
+            FulcioCertificateSigningResponse
+        ] = None
+        self.__signing_certificate: Optional[FulcioCertificateSigningResponse] = None
```

Comment:
Same thought about this: I don't think we need this attribute at all, since in the non-caching case we'll always regenerate it.